pkgs/development/tools/rust/rust-analyzer/test-neovim-lsp.nix

```diff
@@ -1,62 +1,77 @@
-{ runCommand, cargo, neovim, rust-analyzer, rustc }:
-runCommand "test-neovim-rust-analyzer" {
-  nativeBuildInputs = [ cargo neovim rust-analyzer rustc ];
+{
+  runCommand,
+  cargo,
+  neovim,
+  rust-analyzer,
+  rustc,
+}:
+runCommand "test-neovim-rust-analyzer"
+  {
+    nativeBuildInputs = [
+      cargo
+      neovim
+      rust-analyzer
+      rustc
+    ];
 
-  testRustSrc = /* rust */ ''
-    fn main() {
-      let mut var = vec![None];
-      var.push(Some("hello".to_owned()));
-    }
-  '';
+    testRustSrc = # Rust
+      ''
```

Comment:
This inline comment messes up with formatting. Please remove it and nixfmt should format them with one less indentation.

Reply:
These are used for syntax highlighting with treesitter. It's your derivation, so I can... but, I find it helpful for looking at these scripts that are embedded in nix files.

Reply:
> but, I find it helpful for looking at these scripts that are embedded in nix files.

I know that, but comments should not be inserted in an odd location where human will never do. I don't like optimizing machine readability with the cost of human readability. Typically line comments are used like this, which is acceptable for me:
```nix
{
  # Rust
  testRustSrc = ''
    ...
  '';
}
```

If your treesitter highlighting does not recognize this, you can ask upstream to support this hint position. It is possible because I [did it some years ago](https://github.com/oxalica/tree-sitter-nix/blob/master/queries/nvim-injections.scm)[^1] with nvim-treesitter queries.


[^1]: Though I'm not using it now because of the disappointment of their ecosystem, where every editors use a slightly different query language and editor is tightly coupled with a single version of parser implementation of each language.